kernel/Utilities/GlobalMatrix.cpp

```diff
@@ -185,10 +187,23 @@ GlobalPetscMatrix::operator Mat() {
 }
 
 void GlobalPetscMatrix::reinit() {
-    logger.assert_always(rowIndexing_.getMesh() == columnIndexing_.getMesh(),
-                         "Row and column indexing from different meshes");
-    createMat();
-    assemble();
+    if (rowIndexing_.getMesh() == nullptr ||
+        columnIndexing_.getMesh() == nullptr) {
+        // Uninitialized index for at least the rows or columns. So replace it
+        // with an empty matrix.
+        PetscErrorCode error;
+        error = MatDestroy(&A_);
+        CHKERRABORT(PETSC_COMM_WORLD, error);
+        error = MatCreateSeqAIJ(PETSC_COMM_SELF, 0, 0, 0, PETSC_NULL, &A_);
+        CHKERRABORT(PETSC_COMM_WORLD, error);
+        return;
```

Comment:
I thought about wrapping the PETSC matrices in unique_ptr so that they can automatically be destructed by providing a deleter. https://en.cppreference.com/w/cpp/memory/unique_ptr

Reply:
Sounds like an interesting idea for some of the references. However, I think it may be quite a bit more complicated, as depending on how you obtained the PETSc matrix you might need to destroy it in different ways (or not at all).

Reply:
ahh good point. Then nevermind.